dist/videojs-errors.js

```diff
@@ -140,14 +138,16 @@
 
     // PLAYER_ERR_NO_SRC
     player.on('play', function() {
-      if (player.currentSrc() === null ||
-          player.currentSrc() === undefined ||
-          player.currentSrc() === '') {
-        player.error({
-          code: -1,
-          type: 'PLAYER_ERR_NO_SRC'
-        });
-      }
+      window.setTimeout(function(){
+        if (player.currentSrc() === null ||
+            player.currentSrc() === undefined ||
+            player.currentSrc() === '') {
+          player.error({
+            code: -1,
+            type: 'PLAYER_ERR_NO_SRC'
+          });
+        }
+      }, 500);
```

Comment:
Since the above behavior seems strange I agree that this should probably be a video.js fix. I'll reopen this as a new PR without the timeout since karma and using fillWith would be useful.